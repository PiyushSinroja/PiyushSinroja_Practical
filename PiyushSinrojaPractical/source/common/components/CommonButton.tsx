import React from 'react';
import {
  View,
  Text,
  TouchableOpacity
} from 'react-native';

import { CommonButtonStyle } from './CommonButtonStyle';

/**
 * common button inteface used for data type of diclared variables
 *
 * @interface CommonButtonProps
 */
interface CommonButtonProps {
    onPress?(): void;
    titleString: string;
    variationType?: string;
    testId?: string;
    style?: object;
    textStyle?: object;
}

const CommonButton: React.FC<CommonButtonProps> = ({
    onPress,
    testId,
    style,
    textStyle,
    titleString,
  }) => {
    return (
        <View style={CommonButtonStyle.container}>
        <TouchableOpacity
          onPress={onPress}
          testID={testId}
          style={[CommonButtonStyle.buttonContainer, style]}>
          <Text style={[CommonButtonStyle.buttonStyle, textStyle]}>{titleString}</Text>
        </TouchableOpacity>
      </View>
    );
  };
  
export default CommonButton;