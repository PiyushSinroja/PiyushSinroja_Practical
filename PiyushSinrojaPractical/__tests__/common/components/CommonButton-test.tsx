import React from 'react';
import CommonButton from '../../../source/common/components/CommonButton';
import {render, fireEvent} from '@testing-library/react-native';

describe('CommonButton', () => {
    it('CommonButton renders properly', () => {
      const onPressHandler = jest.fn();
      const {toJSON, getByTestId} = render(
        <CommonButton
          testId={'updatedButton'}
          titleString="testing"
          onPress={onPressHandler}
        />,
      );
      fireEvent.press(getByTestId('updatedButton'));
      expect(toJSON()).toMatchSnapshot();
    });
  });