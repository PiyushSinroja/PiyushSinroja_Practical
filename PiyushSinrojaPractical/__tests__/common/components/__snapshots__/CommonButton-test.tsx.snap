// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonButton CommonButton renders properly 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
      "marginBottom": 20,
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "red",
        "borderRadius": 5,
        "height": 80.03999999999999,
        "justifyContent": "center",
        "marginEnd": 20,
        "marginStart": 20,
        "opacity": 1,
      }
    }
    testID="updatedButton"
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 17,
            "fontWeight": "normal",
          },
          undefined,
        ]
      }
    >
      testing
    </Text>
  </View>
</View>
`;
